process/peer/sovereignChainProcess.go

```diff
@@ -50,11 +50,17 @@ func (vs *sovereignChainValidatorStatistics) updateShardDataPeerState(
 		"accumulatedFees", header.GetAccumulatedFees().String(),
 		"developerFees", header.GetDeveloperFees().String(),
 	)
+
+	prevShardData, shardInfoErr := vs.searchInMap(header.GetPrevHash(), cacheMap)
+	if shardInfoErr != nil {
+		return shardInfoErr
+	}
+
 	shardInfoErr = vs.updateValidatorInfoOnSuccessfulBlock(
 		shardConsensus,
-		header.GetPubKeysBitmap(),
-		big.NewInt(0).Sub(header.GetAccumulatedFees(), header.GetDeveloperFees()),
-		header.GetShardID(),
+		prevShardData.GetPubKeysBitmap(),
```

Comment:
For normal process, this update validator info is done with current header data, not previous.

Reply:
Because current header doesn't have all necessary information available like signatures and bitmap, which are only added after the header is sealed.

However, not relevant anymore, I've deleted this func in the next PR, it was a mistake to have it.